src/arch/riscv/RiscvISA.py

```diff
@@ -58,3 +59,18 @@ class RiscvISA(BaseISA):
     riscv_type = Param.RiscvType("RV64", "RV32 or RV64")
 
     enable_rvv = Param.Bool(True, "Enable vector extension")
+
+    def get_isa_string(self):
+        isa_extensions = []
+        # check for the base ISA type
+        if self.riscv_type.value == "RV32":
+            isa_extensions.append("rv32")
+        elif self.riscv_type.value == "RV64":
+            isa_extensions.append("rv64")
+        # use imafdc by default
+        isa_extensions.extend(["i", "m", "a", "f", "d", "c"])
+        # check for the vector extension
+        if self.enable_rvv.value == True:
+            isa_extensions.append("v")
+        isa_string = "".join(isa_extensions)
+        return isa_string
```

Comment:
This code is strange to me... I have a few suggestions (though I believe your code is functionally correct):


```suggestion
    def get_isa_string(self) ->:
        # 32 or 64 bit.
        to_return = "rv32" if self.riscv_type.value == "RV32" else "rv64"
        assert to_return = "rv32 || self.riscv_type.value = "RV64"
        
        # use imafdc by default.
        to_return += "imafdc"
        
        # check for the vector extension
        if self.enable_RVV.value:
            to_return += "v"
        
        return to_return
```

High level:

1. Vector of strings to concatenate strings and characters 🤢.
2. If the case that neither "rv32" and "rv64" are specified is possible, i'd like an error thrown. I think that must be the case here?
3. Nit: `(self.enable_rvv.value == True) == (self.enable_rvv.value)`

I'm guessing you did this `isa_extensions` vector with some grander idea of having the isa be generated automatically from whatever extensions are there, but for now I think i'd prefer the code to be cleaner.

Reply:
Agreed.

Nit

> to_return += "imafdc"

Since each letter represents an extension, I'd like to keep them separately and concat them latter on. I believe this is cleaner than having a hardcoded string.

Reply:
Here's a different suggestion. 

Do we also need to include the Z extension here?

```suggestion
    def get_isa_string(self):
        isa_extensions = []
        # check for the base ISA type
        if self.riscv_type.value == "RV32":
            isa_extensions.append("rv32")
        elif self.riscv_type.value == "RV64":
            isa_extensions.append("rv64")
        # use imafdc by default
        isa_extensions.extend(
            ["i", # integer instructions
            "m", # multiply/divide instructions
            "a", # atomic instructions
            "f", # floating point 
            "d", # double precision floating point
            "c", # compressed instructions
            ]
        )
        # check for the vector extension
        if self.enable_rvv.value == True:
            isa_extensions.append("v") # vector instructions
        isa_string = "".join(isa_extensions)
        return isa_string
```

Reply:
We should include the Z extensions we are supporting. However, I don't think Z extensions that gem5 has need kernel support to be executed correctly.

Also, the V extension is pretty unique in this sense as we need to tell the kernel explicitly.

Having said that, do we have a list of Z extensions that were implemented?

Reply:
Here are the Z exts recognized by the kernel

https://elixir.bootlin.com/linux/v6.5.7/source/arch/riscv/kernel/cpu.c#L213

Reply:
@rogerchang23424 may know the Z extensions we support.

Reply:
gem5 support following Z extensions:
CMO: Zicbom(develop), Zicboz(develop)
Zi*: Zicntr, Zicsr, Zifencei, Zihpm
Zc*: Zca, Zcf, Zcd(includedi in the c exts), Zcb(develop),
Zf*: Zfh
Zb*: Zba, Zbb, Zbc, Zbkb, Zbkc, Zbkx, Zbs
Zk*: Zknd, Zkne, Zknh, Zksed, Zksh

The kernel can recognize the Z exts in v6.5.7 includes:
Zicbom, Zicboz, Zicntr, Zicsr, Zifencei, Zihpm, Zba, Zbb and Zbs

Thanks